sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -22,6 +24,947 @@ namespace Azure.Messaging.EventHubs.Tests
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task StopProcessingSurfacesExceptions(bool async)
+        {
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(15));
+
+            var expectedException = new DivideByZeroException("BOOM!");
+            var mockProcessor = new Mock<EventProcessor<EventProcessorPartition>>(65, "consumerGroup", "namespace", "eventHub", Mock.Of<TokenCredential>(), default(EventProcessorOptions)) { CallBase = true };
+
+            mockProcessor
+                .Setup(processor => processor.CreateConnection())
+                .Throws(expectedException);
+
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
+
+            if (async)
+            {
+                Assert.That(async () => await mockProcessor.Object.StopProcessingAsync(cancellationSource.Token), Throws.Exception.SameAs(expectedException), "The asynchronous close call should bubble the exception.");
+            }
+            else
+            {
+                Assert.That(() => mockProcessor.Object.StopProcessing(cancellationSource.Token), Throws.Exception.SameAs(expectedException), "The asynchronous close call should bubble the exception.");
+            }
+
+            Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation source should not have been triggered.");
+            Assert.That(mockProcessor.Object.IsRunning, Is.False, "The processor should report that it is stopped.");
+            Assert.That(GetRunningProcessorTask(mockProcessor.Object), Is.Null, "There should be no active task for processing.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}.StopProcessing(" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task StopProcessingResetsState(bool async)
+        {
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(15));
+
+            var expectedException = new DivideByZeroException("BOOM!");
+            var mockProcessor = new Mock<EventProcessor<EventProcessorPartition>>(65, "consumerGroup", "namespace", "eventHub", Mock.Of<TokenCredential>(), default(EventProcessorOptions)) { CallBase = true };
+
+            mockProcessor
+                .SetupSequence(processor => processor.CreateConnection())
+                .Throws(expectedException)
+                .Returns(default(EventHubConnection));
+
+            // Starting the processor should result in an exception on the first call, which should leave it in a faulted state.
+
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
```

Comment:
FWIW - I found this behavior to be very surprising.  I would have expected that if `StartProcessingAsync` returned without faulting, then calling `IsRunning` on it would return `true`.

Reply:
It will, unless the fault occurs in the background processing task, in which case it won't be seen by the start call but will be surfaced through the exception handler and potentially stop the processor from running.

In this case, I'm injecting the fault very deliberately for that scenario.   Within the background task after it has been launched, outside visibility of the start method.  We don't have the full implementation here to asset the handler was called;  I expect this will have to change with the next bit of implementation that introduces the full set of processing.

Reply:
For more context, the normal discovery of this would be the error handler being signaled with an exception that isn't tied to a partition context;  the handler should check to see if the processor was still running and make the appropriate decision at that point around restarting or another path.

Open to suggestions if you have a flow that you think makes more sense here; I'm mimicking the `EventProcessorClient` behavior, as I don't have a better idea.

Reply:
> Open to suggestions if you have a flow that you think makes more sense here; I'm mimicking the EventProcessorClient behavior, as I don't have a better idea.

I don't either.  Thanks for explaining how you actually observe the failure through the exception handler in this case.